DOCUMENTATION.md

```diff
@@ -0,0 +1,123 @@
+## Table of Contents
+* [Description](#description)
+* [MQTTClient](#mqttclient)
+* [connect](#connect)
+* [close](#close)
+* [start](#start)
+* [stop](#stop)
+* [setUplinkCallback](#setuplinkcallback)
+  * [uplinkCallback](#uplinkcallback)
+* [setConnectCallback](#setconnectcallback)
+  * [connectCallback](#connectcallback)
+* [setDownlinkCallback](#setpublishcallback)
+  * [downlinkCallback](#publishcallback)
+* [setCloseCallback](#setclosecallback)
+  * [closeCallback](#closecallback)
+* [send](#send)
+* [License](#license)
+
+## Description
+
+This package provides you an easy way to connect to The Things Network via MQTT.
+
+### MQTTClient
+
+The class constructor can be called following this scheme:
+```python
+MQTTClient(appID, appAccessKey, [mqttAddress], [discoveryAddress])
+```
+- `appID`: **string**  this the name given to the application when it was created.
+![Screenshot of the console with app section](./images/app-console.png?raw=true)
+- `appAccessKey`: **string**  this can be found at the bottom of the application page under **ACCESS KEYS**.
+![Screenshot of the console with accesskey section](./images/accesskey-console.png?raw=true)
+- `mqttAddress`: **string**  this  is the address of the handler to which the application was registered. It needs to be provided as an `mqttAddress=value` argument when calling the constructor.
+- `discoveryAddress`: **string** this is the address of the discovery server to use in order to find back the address of the MQTT handler. It needs to be provided as an `discoveryAddress=value` argument when calling the constructor.
+The constructor returns an **MQTTClient object** set up with the application informations, connected to The Things Network.
+
+### connect
+This function connect your client, in case it was closed and need to be openned once again.
+```python
+client.connect()
+```
+
+### close
+Disconnects the MQTT client from which we call the method..
+```python
+client.close()
+```
+
+### start
+Starts an asynchronous loop for the client so that it's possible to run another process (such as a web server) in the same script. This loop is started by default when creating the MQTT client. So it should only be used in case the client was stopped and need to be started again, after re-connecting.
```

Comment:
+ IMHO you could phrase it better. The current phrasing is too long, could be confusing. For example, you could just put:
  > Starts the client in the background.
+ The reconnection is confusing. When do you need to restart is? If the connection is closed, do I use `start()`, or do I use `connect()`? I think you could clarify this.
  + Maybe we could remove either `start()` or `connect()`? As a developer building an application on-top of TTN, I probably am not interested in the difference between the two. Having only one function reduces the client to two states (connected/non-connected), whereas having `start` AND `connect` creates more states (connected+started, connected+not started, non connected+not started). A developer who is interested in the nuance, for some reason, would probably be better off actually using an actual MQTT client, rather than the TTN SDK.
    Was there any specific reason you wanted to keep both the functions?